### PR TITLE
Hide Products (Beta) from inserter

### LINKED ISSUE #43414: Product Collection: hide Products (Beta) from inserter
**Background:**
Products block has been in maintenance mode for some time already. Considering Product Collection covers the functionality of Products (and offers more) and we recommend using it, it makes sense to keep it as a default choice.

**Goals:**
- Hide Products (Beta) block from the inserter
- Add "Products (Beta)" keyword to Product Collection
- Make sure all the existing tests are passing
- Update customer documentation
- Write announcement post

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR hides Products (Beta) from inserter.

This PR will be merged with https://github.com/woocommerce/woocommerce/pull/48112 that replaces Products (Beta) with Product Collection in product archive templates.

Closes https://github.com/woocommerce/woocommerce/issues/43414

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create new post
2. Try adding "Products (Beta)" block by typing:
   - `/products`
   - `/products (beta)`
   - `/products beta`
4. **Expected:** It doesn't appear in the inserter but `Product Collection` does. Order of blocks may differ from the screenshots below

`/products` | `/products (beta)` | `/products beta`
-- | -- | --
image | image | image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: major
+Type: update
+
+Products (Beta): hide block from inserter. Use Product Collection instead
```
Comment (imanish003) [inline plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter]: Nitpick: I believe we can enhance this changelog message.

**Review by imanish003 — APPROVED** [review]
I confirmed that the Products (Beta) block is not visible in the inserter.

I initially thought it would need a lot of testing, but it turns out it doesn't. I'm going to approve this PR. Well done 🚀 :shipit:

File: plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter]: > Nitpick: I believe we can enhance this changelog message.

@imanish003 , sure, what do you suggest? I'm not sure what aspect you have in mind 😅

Comment (imanish003) [reply plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter]: I just didn't liked `dot` in the middle of the sentence 😅

Maybe we can rephrase it:
```
Products (Beta): hide block from inserter in favor of Product Collection block
```

But it was a nitpick, so feel free to ignore it 😅

Comment (kmanijak) [reply plugins/woocommerce/changelog/43414-product-collection-hide-products-beta-from-inserter]: I see and I like that! I updated the changelog.

**Review by thealexandrelara — APPROVED** [review]
Great! Working as expected, so I'm approving it!

**Pull request merged**